packages/runtime-core/src/renderer.ts

```diff
@@ -769,9 +771,10 @@ function baseCreateRenderer(
     isSVG,
     slotScopeIds,
     optimized,
-    start = 0
+    start = 0,
+    end = children.length
   ) => {
-    for (let i = start; i < children.length; i++) {
+    for (let i = start; i < end; i++) {
```

Comment:
Why not make this `end` a local variable rather than add one more argument to the function?